node/pkg/processor/broadcast.go

```diff
@@ -46,18 +50,18 @@ func (p *Processor) broadcastSignature(v *vaa.VAA, signature []byte, txhash []by
 	// Store our VAA in case we're going to submit it to Solana
 	hash := hex.EncodeToString(digest.Bytes())
 
-	if p.state.vaaSignatures[hash] == nil {
-		p.state.vaaSignatures[hash] = &vaaState{
+	if p.state.signatures[hash] == nil {
+		p.state.signatures[hash] = &state{
 			firstObserved: time.Now(),
 			signatures:    map[ethcommon.Address][]byte{},
 			source:        "loopback",
 		}
 	}
 
-	p.state.vaaSignatures[hash].ourVAA = v
-	p.state.vaaSignatures[hash].ourMsg = msg
-	p.state.vaaSignatures[hash].source = v.EmitterChain.String()
-	p.state.vaaSignatures[hash].gs = p.gs // guaranteed to match ourVAA - there's no concurrent access to p.gs
+	p.state.signatures[hash].ourObservation = o
+	p.state.signatures[hash].ourMsg = msg
+	p.state.signatures[hash].source = source
```

Comment:
isn't source still available from the observation?

Reply:
It seems reviewable doesn't update comments directly.  I removed `source` as a parameter for this function in favor of getting it from the observation.

Please re-open if that's not what you meant.